### PR TITLE
feat: impl runInAnonymousContextScope safe for ctxStorage

### PR DESCRIPTION
**Pull request by killagu**
##### Checklist

- [x] `npm test` passes
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message follows commit guidelines

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -622,14 +622,25 @@ declare module 'egg' {
...
-     * Create an anonymous context, the context isn't request level, so the request is mocked.
-     * then you can use context level API like `ctx.service`
-     * @member {String} EggApplication#createAnonymousContext
+     * @Deprecated
+     * It's not safe with app.currentContext, please use app.createAnonymousContextScope() instead
+     * @member {Function} EggApplication#createAnonymousContext
      * @param {Request} req - if you want to mock request like querystring, you can pass an object to this function.
      * @return {Context} context
      */
     createAnonymousContext(req?: Request): Context;
```
Comment (fengmk2) [inline index.d.ts:631]: 加一个 deprecate 日志提醒开发者。

File: lib/egg.js [hunk 2]
```diff
@@ -546,6 +546,11 @@ class EggApplication extends EggCore {
     return this.createContext(request, response);
   }
 
+  createAnonymousContextScope(scope, req) {
```
Comment (fengmk2) [inline lib/egg.js:549]: 这是一个 async function？

File: index.d.ts [hunk 3]
```diff
@@ -622,14 +622,25 @@ declare module 'egg' {
...
 
+    /**
+     * Same as {@link EggApplication#createAnonymousContext},it's safe with app.currentContext.
+     * Create an anonymous context, the context isn't request level, so the request is mocked.
+     * then you can use context level API like `ctx.service`
+     * @member {Function} EggApplication#createAnonymousContext
```
Comment (fengmk2) [inline index.d.ts:637]: EggApplication#createAnonymousContextScope

Comment (fengmk2) [inline index.d.ts:637]: @member {AsyncFunction}

**Comment by killagu** [comment]
不用加了，之前有 runInAnonymousContextScope 这个 api 了。

**Pull request closed without merging**